docs/explanation/charm-architecture.md

```diff
@@ -0,0 +1,18 @@
+# Charm architecture
+
+A [Juju](https://juju.is/) [charm](https://juju.is/docs/olm/charmed-operators) is a set of Python scripts and libraries that operates some software. This charm operates a set of [GitHub self-hosted runners](https://docs.github.com/en/actions/hosting-your-own-runners/managing-self-hosted-runners/about-self-hosted-runners) while handling the security and hardware management.
```

Comment:
The definition of Juju charm seems a bit weird to me

Reply:
s/while handling the security and hardware management/while managing the security and hardware 
Simpler :)

Reply:
I agree we should avoid the `is a set of Python scripts and libraries that operates some software` wording.

Reply:
Also, I don't think this handles "hardware management". We're provisioning on VMs on OpenStack, for instance, so there's no hardware management involved there.

Reply:
> Also, I don't think this handles "hardware management". We're provisioning on VMs on OpenStack, for instance, so there's no hardware management involved there.

Updated to "resource usage".